lib/Service/AssistantService.php

```diff
@@ -153,6 +182,32 @@ public function notifyWhenReady(int $taskId, string $userId): void {
 		}
 	}
 
+	/**
+	 * Cancel notification when task is finished
+	 *
+	 * @param int $taskId
+	 * @param string $userId
+	 * @throws Exception
+	 * @throws MultipleObjectsReturnedException
+	 */
+	public function cancelNotifyWhenReady(int $taskId, string $userId): void {
+		try {
+			$task = $this->taskProcessingManager->getTask($taskId);
+		} catch (NotFoundException $e) {
+			// ignore, task may be already deleted, so delete any dangling notifications
+		} catch (TaskProcessingException $e) {
+			$this->logger->debug('Task request error : ' . $e->getMessage());
+			throw new Exception('Internal server error.', Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		if ($task->getUserId() !== $userId) {
```

Comment:
`$task` might notbe defined here.

Reply:
In which case, delete the task notification anyway? Adding `$task !== null` might be enough in that case.